### PR TITLE
Remove the JSXC/Prosody integration.

### PR DESCRIPTION
**Pull request by skyschub**
As per discussion on https://discourse.diasporafoundation.org/t/removing-diaspora-s-current-chat-integration/2718, nobody raised serious concerns or objections. Given future plans, we do not think having an unfinished implementation of something that likely will not get finished in the current form is worth it. So let's get rid of it.

I deliberately ignored all codestyle warnings. As this PR is exclusively removing things, I felt better not touching anything. Happy to change things if someone feels like it.

When merged, this PR needs documentation in the Changelog that explains the implications.

### CODE REVIEW

File: app/assets/javascripts/app/views/help_view.js [hunk 1]
```diff
@@ -61,9 +53,7 @@ app.views.Help = app.views.StaticContentView.extend({
       title_sharing: Diaspora.I18n.t( 'sharing.title' ),
       title_tags: Diaspora.I18n.t( 'tags.title' ),
       title_keyboard_shortcuts: Diaspora.I18n.t( 'keyboard_shortcuts.title' ),
-      title_miscellaneous: Diaspora.I18n.t( 'miscellaneous.title' ),
-      title_chat: Diaspora.I18n.t( 'chat.title' ),
-      chat_enabled: this.chatEnabled()
+      title_miscellaneous: Diaspora.I18n.t( 'miscellaneous.title' )
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: - Identifier 'title_miscellaneous' is not in camel case.
- There should be no spaces inside this paren.
- Strings must use doublequote.

File: spec/integration/exporter_spec.rb [hunk 2]
```diff
@@ -46,12 +46,10 @@
         user: {
           "contact_groups": [
             {
-              "name":             "generic",
-              "chat_enabled":     false
+              "name":             "generic"
```
Comment (diaspora-code-review) [inline spec/integration/exporter_spec.rb]: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.

File: spec/integration/exporter_spec.rb [hunk 3]
```diff
@@ -46,12 +46,10 @@
...
             },
             {
-              "name":             "Work",
-              "chat_enabled":     false
+              "name":             "Work"
```
Comment (diaspora-code-review) [inline spec/integration/exporter_spec.rb]: Layout/AlignHash: Align the elements of a hash literal if they span more than one line.

File: spec/serializers/export/aspect_serializer_spec.rb [hunk 4]
```diff
@@ -6,8 +6,7 @@
 
   it "has aspect attributes" do
     expect(serializer.attributes).to eq(
-      name:         aspect.name,
-      chat_enabled: aspect.chat_enabled
+      name:         aspect.name
```
Comment (diaspora-code-review) [inline spec/serializers/export/aspect_serializer_spec.rb]: Layout/ExtraSpacing: Unnecessary spacing detected.

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
Overall this looks good to me. I know you ignored codestyle-warnings, but I would still suggest fixing the last 3 warnings about alignment and spacing (I think that improves code quality at these places with only little changes of whitespace), I'm fine with ignoring the first one about camelcase and stuff.

Also, I found the one below, which I think is worth fixing, even not directly related to chat removing.

File: spec/integration/migration_service_spec.rb [hunk 5]
```diff
@@ -148,8 +148,8 @@
           }
         ],
         "contact_groups": [
-          {"name":"Friends","chat_enabled":true},
-          {"name":"Friends","chat_enabled":false}
+          {"name":"Friends"},
+          {"name":"Friends"}
```
[the remark this answers is not in the document]
Comment (denschub) [reply spec/integration/migration_service_spec.rb]: I... have no idea why that's there. Probably an oversight from adding the chat tests and not noticing the names.

File: app/assets/javascripts/app/views/help_view.js [hunk 6]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
-      title_getting_help: Diaspora.I18n.t( 'getting_help.title' ),
-      title_account_and_data_management: Diaspora.I18n.t( 'account_and_data_management.title' ),
-      title_aspects: Diaspora.I18n.t( 'aspects.title' ),
-      title_mentions: Diaspora.I18n.t( 'mentions.title' ),
-      title_pods: Diaspora.I18n.t( 'pods.title' ),
-      title_posts_and_posting: Diaspora.I18n.t( 'posts_and_posting.title' ),
-      title_private_posts: Diaspora.I18n.t( 'private_posts.title' ),
-      title_public_posts: Diaspora.I18n.t( 'public_posts.title' ),
-      title_resharing_posts: Diaspora.I18n.t( 'resharing_posts.title' ),
+      title_header: Diaspora.I18n.t("title_header"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_header' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 7]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_getting_help: Diaspora.I18n.t("getting_help.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_getting_help' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 8]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_account_and_data_management: Diaspora.I18n.t("account_and_data_management.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_account_and_data_management' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 9]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_aspects: Diaspora.I18n.t("aspects.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_aspects' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 10]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_mentions: Diaspora.I18n.t("mentions.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_mentions' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 11]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_pods: Diaspora.I18n.t("pods.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_pods' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 12]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_posts_and_posting: Diaspora.I18n.t("posts_and_posting.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_posts_and_posting' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 13]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_private_posts: Diaspora.I18n.t("private_posts.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_private_posts' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 14]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_public_posts: Diaspora.I18n.t("public_posts.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_public_posts' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 15]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_resharing_posts: Diaspora.I18n.t("resharing_posts.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_resharing_posts' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 16]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
       title_profile: Diaspora.I18n.t("profile.title"),
-      title_sharing: Diaspora.I18n.t( 'sharing.title' ),
-      title_tags: Diaspora.I18n.t( 'tags.title' ),
-      title_keyboard_shortcuts: Diaspora.I18n.t( 'keyboard_shortcuts.title' ),
-      title_miscellaneous: Diaspora.I18n.t( 'miscellaneous.title' ),
-      title_chat: Diaspora.I18n.t( 'chat.title' ),
-      chat_enabled: this.chatEnabled()
+      title_sharing: Diaspora.I18n.t("sharing.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_sharing' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 17]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_tags: Diaspora.I18n.t("tags.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_tags' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 18]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_keyboard_shortcuts: Diaspora.I18n.t("keyboard_shortcuts.title"),
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_keyboard_shortcuts' is not in camel case.

File: app/assets/javascripts/app/views/help_view.js [hunk 19]
```diff
@@ -39,31 +38,22 @@ app.views.Help = app.views.StaticContentView.extend({
...
+      title_miscellaneous: Diaspora.I18n.t("miscellaneous.title")
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/help_view.js]: Identifier 'title_miscellaneous' is not in camel case.

**Review by SuperTux88 — APPROVED** [review]
Thanks for the changes, looks good to me :cookie:

**Comment by SuperTux88** [comment]
Merged to `develop`, thanks :cookie:

**Pull request merged**